apps/rebar/src/rebar3.erl

```diff
@@ -134,7 +134,9 @@ run_aux(State, RawArgs) ->
                      rebar_state:apply_profiles(State, [list_to_atom(Profile)])
              end,
 
-    rebar_utils:check_min_otp_version(rebar_state:get(State1, minimum_otp_vsn, undefined)),
+    MinimumOTPVsn = rebar_state:get(State1, minimum_otp_vsn, undefined),
+    App = rebar_state:current_app(State1),
```

Comment:
Here the app is going to be `undefined` because we're scanning the config before we have even scanned the applications. Do note that the version check in this file may also come from the global configuration file (in `~/.config/rebar3/rebar.config`), in which case any message regarding the application or even the current project might be wrong at this level.

Chances are that since we need to keep the older check's version (see my other comment), we would want to keep using it here when we don't _actually know_ which app configures it or if it's even an app at all.

Reply:
From my reading of the code (and I now see I missed a call to `rebar_app_info:name/1`):

* `rebar_state:current_app/1` returns `undefined | rebar_app_info:t()`
* `rebar_app_info:t()`'s name is `undefined | binary()`
* I can probably just "try and fetch" name from the potentially defined App and let the newly updated code take care of the rest if it's undefined. Is this a complete solution?